src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -808,7 +808,15 @@ private IDelegateCreationExpression CreateBoundDelegateCreationExpressionOperati
         private IMethodReferenceExpression CreateBoundMethodGroupSingleMethodOperation(BoundMethodGroup boundMethodGroup, IMethodSymbol methodSymbol, bool suppressVirtualCalls)
         {
             bool isVirtual = (methodSymbol.IsAbstract || methodSymbol.IsOverride || methodSymbol.IsVirtual) && !suppressVirtualCalls;
-            Lazy<IOperation> instance = new Lazy<IOperation>(() => Create(boundMethodGroup.InstanceOpt));
+            BoundExpression receiverOpt = boundMethodGroup.InstanceOpt;
+
+            if (methodSymbol.IsStatic && receiverOpt != null && 
+                receiverOpt.WasCompilerGenerated && receiverOpt.Kind == BoundKind.ThisReference)
```

Comment:
Why are we generating a `ThisReference` in the bound tree if the method is static?

Reply:
The receiver is coming from a MethodGroup, I guess compiler optimistically synthesizes ```ThisReference``` before overload resolution takes place.